hscontrol/types/acl.go

```diff
@@ -0,0 +1,45 @@
+package types
+
+import (
+	"encoding/json"
+	"errors"
+	"time"
+
+	"google.golang.org/protobuf/types/known/structpb"
+	"gorm.io/datatypes"
+
+	v1 "github.com/juanfont/headscale/gen/go/headscale/v1"
+)
+
+var (
+	ErrACLPolicyNotFound         = errors.New("acl policy not found")
+	ErrInvalidACLPolicyFormat    = errors.New("invalid policy format")
+	ErrACLPolicyUpdateIsDisabled = errors.New("update is disabled for modes other than 'db'")
+)
+
+// ACL describes the data model for ACLs used to restrict access to resources.
+type ACL struct {
+	ID     uint64 `gorm:"primary_key"`
+	Policy datatypes.JSON
+
+	CreatedAt  *time.Time
+	Expiration *time.Time
```

Comment:
Is `Expiration` and `LastSeen` copy pasta? :-)

Reply:
Yes. 😅 I'll fix this.

Reply:
GORM also  adds created at I believe, so that can also be omitted